src/odemis/gui/cont/tabs.py

```diff
@@ -2301,91 +2304,158 @@ def __init__(self, name, button, panel, main_frame, main_data):
+            panel.btn_switch_milling.Bind(wx.EVT_BUTTON, self._on_switch_btn)
+            panel.btn_switch_coating.Bind(wx.EVT_BUTTON, self._on_switch_btn)
+            panel.stage_align_btn_p_aligner_x.Bind(wx.EVT_BUTTON, self._on_aligner_btn)
+            panel.stage_align_btn_m_aligner_x.Bind(wx.EVT_BUTTON, self._on_aligner_btn)
+            panel.stage_align_btn_p_aligner_y.Bind(wx.EVT_BUTTON, self._on_aligner_btn)
+            panel.stage_align_btn_m_aligner_y.Bind(wx.EVT_BUTTON, self._on_aligner_btn)
+            panel.stage_align_btn_p_aligner_z.Bind(wx.EVT_BUTTON, self._on_aligner_btn)
+            panel.stage_align_btn_m_aligner_z.Bind(wx.EVT_BUTTON, self._on_aligner_btn)
+            panel.btn_cancel.Bind(wx.EVT_BUTTON, self._on_cancel)
+
+            # Determine and show current position of the stage
+            self._current_position, self._target_position = None, None
+            self._enable_movement_controls()
+            if self._current_position in self.position_btns.keys():
+                pos_button = next(button for pos, button in self.position_btns.items() if pos == self._current_position)
+                self._toggle_switch_buttons(pos_button)
+            self._show_cancel_warning_msg(None)
+
+        elif self._role == 'meteor':
+            self._stage  = self.tab_data_model.main.stage_bare
+            # start and end position are used for the gauge progress bar
+            self._start_pos = self._end_pos = None
+            # Show current position of the stage via the progress bar
+            self._stage.position.subscribe(self._update_progress_bar, init=False)
+            self._stage.position.subscribe(self._on_stage_pos)
+            # metadata 
+            # check for the metadata of meteor stage positions
+            if not model.MD_SAMPLE_CENTERS in self._stage.getMetadata():
+                raise ValueError("The stage misses the 'SAMPLE_CENTERS' metadata.")
+            if not model.MD_FAV_POS_DEACTIVE in self._stage.getMetadata():
+                raise ValueError("The stage misses the 'FAV_POS_DEACTIVE' metadata.")
+            if not model.MD_SEM_IMAGING_RANGE in self._stage.getMetadata():
+                raise ValueError("The stage misses the 'SEM_IMAGING_RANGE' metadata.")
+            if not model.MD_FM_IMAGING_RANGE in self._stage.getMetadata():
+                raise ValueError("The stage misses the 'FM_IMAGING_RANGE' metadata.")
+            if not model.MD_POS_COR in self._stage.getMetadata():
+                raise ValueError("The stage misses the 'POS_COR' metadata.")
+            # Fail early when required axes are not found on the positions metadata
+            focuser = self.tab_data_model.main.focus
+            focus_md = focuser.getMetadata()
+            # take only the active and deactive position of the focuser
+            for md in focus_md.keys():
+                if md not in [model.MD_FAV_POS_DEACTIVE, model.MD_FAV_POS_ACTIVE]:
+                    focus_md.pop(md)
```

Comment:
I don't think you can remove from the dictionary during iteration (try it). I believe you could iterate the dict items so you have both the keys & value in this manner:
```
for f_key, f_value in focus_md.items():
      if f_key in [model.MD_FAV_POS_DEACTIVE, model.MD_FAV_POS_ACTIVE] & not required_axis.issubset(f_value.keys()):
```

Reply:
Mmmmh, I added an additional metadata without the z axis, it gave me this error:
`RuntimeError: dictionary changed size during iteration`

Reply:
I see. When i tested i changed the axis from z to x. This is the error i get which is the expected. 
![error](https://user-images.githubusercontent.com/79858941/131855888-85c42024-b231-4ccf-86c3-55b590349871.png)

Reply:
Apparently, i only tested the part that checks for the required axis, not the part that pops/removes the unwanted metadata.